lib/system.nim

```diff
@@ -2566,7 +2566,7 @@ proc `[]`*[T, U, V](s: openArray[T], x: HSlice[U, V]): seq[T] =
   newSeq(result, L)
   for i in 0 ..< L: result[i] = s[i + a]
 
-proc `[]=`*[T, U, V](s: var seq[T], x: HSlice[U, V], b: openArray[T]) =
+proc `[]=`*[T; U: Ordinal, V: Ordinal](s: var seq[T], x: HSlice[U, V], b: openArray[T]) =
```

Comment:
future work:
note that this relies on some undocumented feature that we should document in manual.rst https://nim-lang.github.io/Nim/manual.html#procedures only mentions proc params, not generic params:

> The type applies to all parameters immediately before it, until either the beginning of the parameter list, a semicolon separator, or an already typed parameter, is reached.